### PR TITLE
add retest doc

### PR DESCRIPTION
**Pull request by carlory**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

If the failure is a flake, it needs to be manually set out by @RainbowMango to rerun tests.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by XiShanYongYe-Chang — COMMENTED** [review]
It looks clear, Thanks!

/cc @Tingtal @RainbowMango

**Comment by Tingtal** [comment]
Looks good to me too. :D

**Comment by carlory** [comment]
PTAL @RainbowMango

**Comment by carlory** [comment]
@XiShanYongYe-Chang @Tingtal @RainbowMango @kevin-wangzefeng 

 could you give me a `/lgtm` or `/approve` label on this pr ?

**Comment by carlory** [comment]
Hi @RainbowMango , any suggestions here?

**Comment by carlory** [comment]
Please take a look @RainbowMango

**Review by RainbowMango — COMMENTED** [review]
The `The Testing and Merge Workflow` is part of `PR workflow`, I'd prefer to add a `pull-requests.md` just like Kubernetes :
https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md

File: docs/contributors/guide/github-workflow.md [hunk 1]
```diff
@@ -277,3 +277,31 @@ git push ${your_remote_name} myrevert
...
+
+1. The CI workflow will run:
+
+    1. Automatic tests run. See the current list of tests at this [link](https://github.com/karmada-io/karmada/actions/workflows/ci.yml)
+    1. If tests fail, resolve issues by pushing edits to your pull request branch
+    1. If the failure is a flake, use the following command on your local pull request branch to rerun all tests
+        ```shell
+        git commit --allow-empty --signoff --amend --no-edit
+        git push --force
+        ```
```
[the remark this answers is not in the document]
Comment (carlory) [reply docs/contributors/guide/github-workflow.md:296]: ok

**Pull request closed without merging**